from0k2bp.tex

```diff
@@ -959,85 +977,81 @@
+\item $\mathcal{P}$ → $\mathcal{V}$: $R$ (a new random curve point, but $\mathcal{P}$ knows $k$ s.t. $R=kG$)
+\item $\mathcal{V}$ → $\mathcal{P}$: $e$ (a random scalar)
+\item $\mathcal{P}$ → $\mathcal{V}$: $s$ (which $\mathcal{P}$ calculated from the equation $s=k+ex$), and
+\item $\mathcal{P}$ → $\mathcal{V}$: $C_0$ (a new commitment to a newly chosen random vector of dimension $N$)
+\item $\mathcal{V}$ → $\mathcal{P}$: $e$ (a random scalar)
+\item $\mathcal{P}$ → $\mathcal{V}$: $(\mathbf{z}, s)$ (a single vector of dimension $N$, and another scalar). 
+\end{itemize}
 
-(the first was Schnorr's identity protocol; the second was the proof of
-knowledge of a set of vectors). These are both examples of Sigma
+These are both examples of Sigma
 protocols, so called because of a vague resemblance to the greek letter
 $\Sigma$, in that the process goes forwards once, then backwards, then forwards
 finally. The common pattern, though, is more than this three step
 interactive process. We generalise it as something like:
 
-P → V: commitment
-
-V → P: challenge
-
-P → V: response (proof)
+\begin{itemize}
+\item $\mathcal{P}$ → $\mathcal{V}$: commitment
+\item $\mathcal{V}$ → $\mathcal{P}$: challenge
+\item $\mathcal{P}$ → $\mathcal{V}$: response (proof)
+\end{itemize}
 
 It's worth emphasizing this structure, and why it arises, at this point
-in the discourse, so the mathematical structure of the inner product
-proof isn't too overwhelming. Consider the Schnorr case (basically the
+in the discourse. It makes the mathematical structure of the inner product
+proof less overwhelming then. Consider the Schnorr case (basically the
 simplest) to see why it's necessary for the provision of a proof. Say
-you have a secret value x.
+you have a secret value $x$.
 
-\underline{Option 1: just send $x$ to V}
+\underline{Option 1: just send $x$ to $\mathcal{V}$}
 
 This would be fine except we're trying to keep $x$ secret, so \ldots{}
 
-\underline{Option 2: send to V}
+\underline{Option 2: send to $\mathcal{V}$}
 
 This hides (``blinds'' is sometimes the term used) $x$ perfectly (in
-particular, to an outside observer of the communcation, so that's
-something), but it also leaves V with no way of verifying.
+particular, to an outside observer of the communication, so that's
+something), but it also leaves $\mathcal{V}$ with no way of verifying.
 
-The first attempt to get out of this impasse is to let V get involved --
+The first attempt to get out of this impasse is to let $\mathcal{V}$ get involved --
 interactivity. Hence the ``challenge''.
 
-\underline{Option 3: V sends challenge $e$, P sends $x+e$}
+\underline{Option 3: $\mathcal{V}$ sends challenge $e$, $\mathcal{P}$ sends $x+e$}
 
 Dumb of course; this doesn't hide anything from V since V knows $e$.
 
 At this point we would just be stuck and give up, except for one thing:
-we have \emph{one-way-functions}. This means P can send a
+we have \emph{one-way functions}. This means $\mathcal{P}$ can send a
```

Comment:
second hyphen shouldn't be used.

Reply:
I'm not sure what you mean. Is the proposed change correct or not?

Reply:
Well imo "one-way functions" is correct and "one-way-functions" is not correct. To back up my statement, look at the page for this topic on wikipedia and on mathworld.wolfram.com

Reply:
Well, yes. That's why I proposed the change to `one-way functions`.

Reply:
Ah sorry; reading fail.